internal/connect/command_offset_describe.go

```diff
@@ -0,0 +1,178 @@
+	"github.com/confluentinc/cli/v3/pkg/kafka"
+	"github.com/confluentinc/cli/v3/pkg/output"
+	"github.com/confluentinc/cli/v3/pkg/retry"
+)
+
+type humanOffsetConnectorOut struct {
+	Id       string `human:"ID"`
+	Name     string `human:"Name"`
+	Offsets  string `human:"Offsets"`
+	Metadata string `human:"Metadata"`
+}
+
+type serializedOffsetConnectorOut struct {
+	Id       string                                      `json:"id" yaml:"id"`
+	Name     string                                      `json:"name" yaml:"name"`
+	Offsets  []map[string]any                            `json:"offsets" yaml:"offsets"`
+	Metadata connectv1.ConnectV1ConnectorOffsetsMetadata `json:"metadata" yaml:"metadata"`
+}
+
+func (c *offsetCommand) newDescribeCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "describe <id>",
+		Short:             "Describe connector offsets.",
+		Args:              cobra.ExactArgs(1),
+		RunE:              c.describe,
+		ValidArgsFunction: pcmd.NewValidArgsFunction(c.validArgs),
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Describe offsets for a connector in the current or specified Kafka cluster context.",
+				Code: "confluent connect offset describe lcc-123456",
+			},
+			examples.Example{
+				Code: "confluent connect offset describe lcc-123456 --cluster lkc-123456",
+			},
+		),
+	}
+
+	cmd.Flags().Int32("staleness-threshold", 120, "Repeatedly fetch offsets until we get an offset with an observed time within staleness threshold in seconds. (min 5)")
+	cmd.Flags().Int32("refetch-timeout", 30, "Max time in seconds to wait until we get an offset within the staleness threshold.")
+
+	pcmd.AddClusterFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddEnvironmentFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddOutputFlag(cmd)
+
+	return cmd
+}
+
+func (c *offsetCommand) describe(cmd *cobra.Command, args []string) error {
+	kafkaCluster, err := kafka.GetClusterForCommand(c.V2Client, c.Context)
+	if err != nil {
+		return err
+	}
+
+	environmentId, err := c.Context.EnvironmentId()
+	if err != nil {
+		return err
+	}
+
+	connector, err := c.V2Client.GetConnectorExpansionById(args[0], environmentId, kafkaCluster.ID)
+	if err != nil {
+		return err
+	}
+
+	connectorName := connector.Info.GetName()
+	offsets, err := c.V2Client.GetConnectorOffset(connector.Info.GetName(), environmentId, kafkaCluster.ID)
+	if err != nil {
+		return err
+	}
+	stalenessThreshold, err := cmd.Flags().GetInt32("staleness-threshold")
+	if err != nil {
+		return err
+	}
+
+	if stalenessThreshold < 5 {
+		return fmt.Errorf("staleness-threshold cannot be less than 5 seconds")
+	}
+
+	refetchTimeout, err := cmd.Flags().GetInt32("refetch-timeout")
+	if err != nil {
+		return err
+	}
+
+	var apiErr error
+	if offsets.HasMetadata() && offsets.Metadata.HasObservedAt() && int32(time.Since(*offsets.Metadata.ObservedAt).Seconds()) > stalenessThreshold {
```

Comment:
Can you check if time.Since automatically converts to UTC

Reply:
Yes